src/lib/y2storage/proposal/planned_devices_generator.rb

```diff
@@ -98,38 +100,44 @@ def additional_devices
 
       # Device to host the swap space according to the settings.
       def swap_device
-        swap_size = DEFAULT_SWAP_SIZE
+        min_size = max_size = MAX_SWAP_SIZE
         if settings.enlarge_swap_for_suspend
-          swap_size = [ram_size, swap_size].max
+          min_size = max_size = [ram_size, max_size].max
+        elsif @target == :min
+          min_size = MIN_SWAP_SIZE
+          # leave max_size alone so any leftover free space can be distributed
+          # to swap, too according to SWAP_WEIGHT
         end
         if settings.use_lvm
-          swap_lv(swap_size)
+          swap_lv(min_size, max_size)
         else
-          swap_partition(swap_size)
+          swap_partition(min_size, max_size)
         end
       end
 
-      def swap_lv(size)
+      def swap_lv(min_size, max_size)
         lv = Planned::LvmLv.new("swap", Filesystems::Type::SWAP)
         lv.logical_volume_name = "swap"
-        lv.min_size = size
-        lv.max_size = size
+        lv.min_size = min_size
+        lv.max_size = max_size
+        lv.weight = SWAP_WEIGHT
         lv
       end
 
-      def swap_partition(size)
+      def swap_partition(min_size, max_size)
         part = Planned::Partition.new("swap", Filesystems::Type::SWAP)
         part.encryption_password = settings.encryption_password
         # NOTE: Enforcing the re-use of an existing partition limits the options
         # to propose a valid distribution of the planned partitions. For swap we
         # already have mechanisms to reuse UUIDs and labels, so maybe is smarter
         # to never reuse partitions as-is.
-        reuse = reusable_swap(size)
+        reuse = reusable_swap(max_size)
```

Comment:
NP: is debatable if we should use `max_size` or `min_size` here. But I don't care much. As the comment states, I hope we will remove this reusing anyway in the future.